### PR TITLE
[move-2024]Use some Move 2024 Syntax to update 'defi' example

### PR DESCRIPTION
**Pull request by shaflow01**
## Description 

Use some Move 2024 Syntax to update 'defi' example .And more convenient method calls.

## Test plan 

I built locally and allowed testing, and they all succeeded without warning
test

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

**Comment by shaflow01** [comment]
@stefan-mysten I don't know how to apply for review from the code owner. Could you guide me on how to do it?

**Comment by stefan-mysten** [comment]
Thanks for your contribution @shaflow01. For `Move` related code, you could tag this to `devx-move`, which includes all the Move team. 
Thanks!

File: sui_programmability/examples/defi/sources/flash_lender.move [hunk 1]
```diff
@@ -66,7 +66,7 @@ module defi::flash_lender {
     /// current transaction.
     public fun new<T>(to_lend: Balance<T>, fee: u64, ctx: &mut TxContext): AdminCap {
         let id = object::new(ctx);
-        let flash_lender_id = object::uid_to_inner(&id);
+        let flash_lender_id = id.uid_to_inner();
```
Comment (amnn) [inline sui_programmability/examples/defi/sources/flash_lender.move:69]: I believe there is a more terse `to_inner` alias for this one.

```suggestion
        let flash_lender_id = id.to_inner();
```

File: sui_programmability/examples/defi/sources/pool.move [hunk 2]
```diff
@@ -578,4 +578,4 @@ module defi::pool_tests {
     // utilities
     fun scenario(): Scenario { test::begin(@0x1) }
     fun people(): (address, address) { (@0xBEEF, @0x1337) }
-}
+}
```
Comment (amnn) [inline sui_programmability/examples/defi/sources/pool.move:581]: nit: Could you add back the trailing newlines?

**Review by amnn — COMMENTED** [review]
Thanks for the contribution @shaflow01 ! This looks good. There's just a couple of nits here and there, if you could take a look!

**Comment by shaflow01** [comment]
@amnn Could you please check if my modifications are correct?

File: sui_programmability/examples/defi/sources/escrow.move [hunk 3]
```diff
@@ -85,7 +85,7 @@ module defi::escrow {
         let EscrowedObj {
             id, sender, recipient: _, exchange_for: _, escrowed
         } = obj;
-        object::delete(id);
+        id.delete();
         transfer::public_transfer(escrowed, sender)
     }
-}
+}
```
Comment (amnn) [inline sui_programmability/examples/defi/sources/escrow.move:91]:
```suggestion
}

```

File: sui_programmability/examples/defi/sources/flash_lender.move [hunk 4]
```diff
@@ -172,4 +172,4 @@ module defi::flash_lender {
     public fun flash_lender_id<T>(self: &Receipt<T>): ID {
         self.flash_lender_id
     }
-}
+}
```
Comment (amnn) [inline sui_programmability/examples/defi/sources/flash_lender.move:175]:
```suggestion
}

```

File: sui_programmability/examples/defi/sources/pool.move [hunk 5]
```diff
@@ -577,5 +577,6 @@ module defi::pool_tests {
 
     // utilities
     fun scenario(): Scenario { test::begin(@0x1) }
+    
     fun people(): (address, address) { (@0xBEEF, @0x1337) }
-}
+}
```
Comment (amnn) [inline sui_programmability/examples/defi/sources/pool.move:582]:
```suggestion
}

```

File: sui_programmability/examples/defi/sources/shared_escrow.move [hunk 6]
```diff
@@ -63,8 +63,8 @@ module defi::shared_escrow {
         escrow: &mut EscrowedObj<T, ExchangeForT>,
         ctx: &TxContext
     ) {
-        assert!(&tx_context::sender(ctx) == &escrow.creator, EWrongOwner);
+        assert!(&ctx.sender() == &escrow.creator, EWrongOwner);
         assert!(option::is_some(&escrow.escrowed), EAlreadyExchangedOrCancelled);
         transfer::public_transfer(option::extract<T>(&mut escrow.escrowed), escrow.creator);
     }
-}
+}
```
Comment (amnn) [inline sui_programmability/examples/defi/sources/shared_escrow.move:70]:
```suggestion
}

```

File: sui_programmability/examples/defi/sources/subscription.move [hunk 7]
```diff
@@ -161,4 +161,4 @@ module defi::smart_swapper {
         // do something with swapped values ?
         // transfer::public_transfer( ... )
     }
-}
+}
```
Comment (amnn) [inline sui_programmability/examples/defi/sources/subscription.move:164]:
```suggestion
}

```

File: sui_programmability/examples/defi/tests/escrow_tests.move [hunk 8]
```diff
@@ -157,11 +157,11 @@ module defi::escrow_tests {
...
-        test_scenario::next_tx(scenario, BOB_ADDRESS);
+        scenario.next_tx(BOB_ADDRESS);
         new_scenario
     }
 
     fun owns_object<T: key + store>(owner: address): bool {
         test_scenario::has_most_recent_for_address<T>(owner)
     }
-}
+}
```
Comment (amnn) [inline sui_programmability/examples/defi/tests/escrow_tests.move:167]:
```suggestion
}

```

File: sui_programmability/examples/defi/tests/flash_lender_tests.move [hunk 9]
```diff
@@ -57,8 +57,8 @@ module defi::flash_lender_tests {
...
 
             test_scenario::return_shared(lender_val);
-            test_scenario::return_to_sender(scenario, admin_cap);
+            scenario.return_to_sender(admin_cap);
         };
-        test_scenario::end(scenario_val);
+        scenario_val.end();
     }
-}
+}
```
Comment (amnn) [inline sui_programmability/examples/defi/tests/flash_lender_tests.move:64]:
```suggestion
}

```

File: sui_programmability/examples/defi/tests/shared_escrow_test.move [hunk 10]
```diff
@@ -176,11 +176,11 @@ module defi::shared_escrow_tests {
...
-        test_scenario::next_tx(scenario, alice);
+        scenario.next_tx(alice);
         (new_scenario, item_b)
     }
 
     fun owns_object<T: key + store>(owner: address): bool {
         test_scenario::has_most_recent_for_address<T>(owner)
     }
-}
+}
```
Comment (amnn) [inline sui_programmability/examples/defi/tests/shared_escrow_test.move:186]:
```suggestion
}

```

**Review by amnn — COMMENTED** [review]
The trailing newlines are still now showing up unfortunately -- let me see if I can amend them using suggestions.

**Comment by shaflow01** [comment]
@amnn Sorry about didn't understand your meaning before.
How long will they be merged ? Due to lack of experience, I used the main branch to submit this push request

**Review by amnn — APPROVED** [review]
Sorry for the delay, this is good to go! Thanks for the continued contribution @shaflow01 !

**Pull request merged**